clicache/src/PoolFactory.hpp

```diff
@@ -279,6 +279,20 @@ namespace Apache
         /// </exception>
         PoolFactory^ AddServer(String^ host, Int32 port);
 
+        /// <summary>
+        /// Set proxy info for SNI connection.
+        /// </summary>
+        /// <remarks>
+        /// Used for connecting via SNI proxy.
+        /// </remarks>
+        /// <param>
+        /// host the host name or ip address that the server is listening on.
+        /// </param>
+        /// <param>
+        /// port the port that the server is listening on.
+        /// </param>
+        PoolFactory^ SetSniProxy(String^ hostname, Int32 port);
```

Comment:
Deviates from [SNI Proxy RFC](https://cwiki.apache.org/confluence/display/GEODE/Client+side+configuration+for+a+SNI+proxy).

Reply:
Native impl of SNI is consistent with Native's libraries... Java client impl of SNI simply doesn't map to this API.
Our understanding is that the light-weight RFC process exists to avoid issues post implementation etc or any surprises... Native's API for SNI does differ from the Java client, but the core feature etc is in line.

Reply:
A difference in API is a pretty big surprise to some. Especially when you consider that .NET does expose a Socket concept in the SDK similar to Java. Having a discussion beforehand as to why one API would deviate from another established in an RFC helps reviewers understand the rational for the deviation.

I can only make the assumption that the deviation is a result of the C++ layer not wanting to leak `ACE::Socket` outside the internals. There are alternative ways to do that such that you have an API similar to the Java API that is extensible for future proxy implementations or parameters to the proxy connections itself without exposing the internals of the socket implementation. By not having this conversation we have implemented something that solves today's issue but will need to be deprecated or augmented to support future changes to the proxy strategy.

Reply:
not leaking anything from `ACE` out to the public API is important - especially since we are close to removing all of `ACE` from this code base...

Reply:
So why not provide, like the Java API, and interface that can take pre-canned `SocketFactory` classes. At this point the only deviation would be the ability for users to provide their own factories?

Reply:
Offline conversation: The team is comfortable with the limitations of this API in the effort to get something released and are comfortable with potentially deprecating this API in favor of something more flexible and in line with the RFC when and if expansion of this feature is necessary.